google-cloud-spanner/src/main/java/com/google/cloud/spanner/Options.java

```diff
@@ -1052,4 +1072,30 @@ public boolean equals(Object o) {
       return o instanceof LastStatementUpdateOption;
     }
   }
+
+  static final class RequestIdOption extends InternalOption
+      implements ReadOption, TransactionOption, UpdateOption {
+    private final XGoogSpannerRequestId reqId;
+
+    RequestIdOption(XGoogSpannerRequestId reqId) {
+      this.reqId = reqId;
+    }
+
+    @Override
+    void appendToOptions(Options options) {
+      options.reqId = this.reqId;
+    }
+
+    @Override
+    public int hashCode() {
+      return RequestIdOption.class.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      // TODO: Examine why the precedent for LastStatementUpdateOption
+      // does not check against the actual value.
```

Comment:
The reason for that is that `LastStatementUpdateOption` does not contain any state. It basically means that `whenever you see this option, you should always add `LastStatement=true` to the request'.

Reply:
Thank you for the explanation!